autoload/ale/statusline.vim

```diff
@@ -60,53 +53,6 @@ function! s:GetCounts(buffer) abort
         call ale#statusline#Update(a:buffer, g:ale_buffer_info[a:buffer].loclist)
     endif
 
-    return g:ale_buffer_info[a:buffer].count
-endfunction
-
-" Returns a Dictionary with counts for use in third party integrations.
-function! ale#statusline#Count(buffer) abort
-    " The Dictionary is copied here before exposing it to other plugins.
-    return copy(s:GetCounts(a:buffer))
-endfunction
-
-" This is the historical format setting which could be configured before.
-function! s:StatusForListFormat() abort
-    let [l:error_format, l:warning_format, l:no_errors] = g:ale_statusline_format
-    let l:counts = s:GetCounts(bufnr(''))
-
-    " Build strings based on user formatting preferences.
-    let l:errors = l:counts[0] ? printf(l:error_format, l:counts[0]) : ''
-    let l:warnings = l:counts[1] ? printf(l:warning_format, l:counts[1]) : ''
-
-    " Different formats based on the combination of errors and warnings.
-    if empty(l:errors) && empty(l:warnings)
-        let l:res = l:no_errors
-    elseif !empty(l:errors) && !empty(l:warnings)
-        let l:res = printf('%s %s', l:errors, l:warnings)
-    else
-        let l:res = empty(l:errors) ? l:warnings : l:errors
-    endif
-
-    return l:res
-endfunction
-
-" Returns a formatted string that can be integrated in the statusline.
-"
-" This function is deprecated, and should not be used. Use the airline plugin
-" instead, or write your own status function with ale#statusline#Count()
-function! ale#statusline#Status() abort
```

Comment:
These functions are deprecated, but we have to keep them for now. After we release version 1.9, we can start removing deprecated features for version 2.0. We should mark some more code as deprecated first, like any remaining uses of `_args` options, and emit warnings for each.